src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +78,268 @@ public class AnalysisService : IDisposable
+        /// Sets up a script analysis run, eventually returning the result.
         /// </summary>
-        /// <param name="enable">Enable the rule.</param>
-        /// <param name="exportedOnly">Analyze only exported functions/cmdlets.</param>
-        /// <param name="blockComment">Use block comment or line comment.</param>
-        /// <param name="vscodeSnippetCorrection">Return a vscode snipped correction should be returned.</param>
-        /// <param name="placement">Place comment help at the given location relative to the function definition.</param>
-        /// <returns>A PSScriptAnalyzer settings hashtable.</returns>
-        public static Hashtable GetCommentHelpRuleSettings(
-            bool enable,
-            bool exportedOnly,
-            bool blockComment,
-            bool vscodeSnippetCorrection,
-            string placement)
-        {
-            var settings = new Dictionary<string, Hashtable>();
-            var ruleSettings = new Hashtable();
-            ruleSettings.Add("Enable", enable);
-            ruleSettings.Add("ExportedOnly", exportedOnly);
-            ruleSettings.Add("BlockComment", blockComment);
-            ruleSettings.Add("VSCodeSnippetCorrection", vscodeSnippetCorrection);
-            ruleSettings.Add("Placement", placement);
-            settings.Add("PSProvideCommentHelp", ruleSettings);
-            return GetPSSASettingsHashtable(settings);
-        }
-
-        /// <summary>
-        /// Construct a PSScriptAnalyzer settings hashtable
-        /// </summary>
-        /// <param name="ruleSettingsMap">A settings hashtable</param>
-        /// <returns></returns>
-        public static Hashtable GetPSSASettingsHashtable(IDictionary<string, Hashtable> ruleSettingsMap)
+        /// <param name="filesToAnalyze">The files to run script analysis on.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel this call with.</param>
+        /// <returns>A task that finishes when script diagnostics have been published.</returns>
+        public void RunScriptDiagnostics(
+            ScriptFile[] filesToAnalyze,
+            CancellationToken cancellationToken)
         {
-            Validate.IsNotNull(nameof(ruleSettingsMap), ruleSettingsMap);
-
-            var hashtable = new Hashtable();
-            var ruleSettingsHashtable = new Hashtable();
+            if (AnalysisEngine == null)
+            {
+                return;
+            }
 
-            hashtable["IncludeRules"] = ruleSettingsMap.Keys.ToArray<object>();
-            hashtable["Rules"] = ruleSettingsHashtable;
+            // Create a cancellation token source that will cancel if we do or if the caller does
+            var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
 
-            foreach (var kvp in ruleSettingsMap)
+            // If there's an existing task, we want to cancel it here;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)

Reply:
you're drunk codacy